### PR TITLE
tls13-mlkem: verify that the server verifies correctness of the whole key

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Verify that all the encoded matrix variables are checked for correctness

### Motivation and Context

More comprehensive testing of ML-KEM

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tlsfuzzer/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

- - -
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/968)

### CODE REVIEW

File: scripts/test-tls13-mlkem.py [hunk 1]
```diff
@@ -62,7 +63,7 @@ def help_msg():
 def main():
     host = "localhost"
     port = 4433
-    num_limit = None
+    num_limit = 400
```
Comment (ep69) [inline scripts/test-tls13-mlkem.py:66]: Later (line 421), there is a check `if not num_limit`. Do we want to keep that for cases when the test is invoked with `-n 0`?

**Review by ep69 — DISMISSED** [review]
See my comments, just for sure.

File: scripts/test-tls13-mlkem.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-tls13-mlkem.py:66]: precisely that, yes

File: scripts/test-tls13-mlkem.py [hunk 2]
```diff
@@ -289,41 +290,63 @@ def main():
...
-            key_shares = [key_share_gen(group)]
+            # last 32 bytes of the public key are "rho" in ML-KEM; i.e.
+            # seed that can have any values; change only values of the t_hat
             if group == GroupName.x25519mlkem768:
-                key_shares[0].key_exchange[1] = 0xff
+                pqc_start = 0
+                pqc_length = 384 * 3
             elif group == GroupName.secp256r1mlkem768:
-                key_shares[0].key_exchange[67] = 0xff
+                pqc_start = 65
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-mlkem.py:299]: the original numbers were arbitrary (first ones that made the server consistently fail in my quick testing)

the numbers now are from the definition: like secp256r1 has key shares that are `(256/8) * 2 + 1` bytes long (256 bits per point coordinate + 1 byte for point format), same for P-384

**Review by ep69 — APPROVED** [review]

**Pull request merged**